samples/dymaptic.GeoBlazor.Core.Sample.Shared/Pages/BingMapsLayers.razor

```diff
@@ -0,0 +1,84 @@
+﻿@page "/bingmaps-layer"
+<PageTitle>BingMapsLayer</PageTitle>
+<h1>Bing Maps Layer</h1>
+<div class="links-div">
+    <a class="btn btn-secondary" target="_blank" href="https://developers.arcgis.com/javascript/latest/api-reference/esri-layers-BingMapsLayer.html">ArcGIS API for JavaScript Reference</a>
+    <a class="btn btn-primary" target="_blank" href="https://www.microsoft.com/en-us/maps/bing-maps/choose-your-bing-maps-api">Bing Maps API</a>
+</div>
+<p class="instructions">
+    A simple demo displaying a Bing Maps Layer. You can generate a free Bing Maps Key <a target="_blank" href="https://www.bingmapsportal.com/">here</a>.
+</p>
+
+<div class="form-group">
+    <Label>Enter a valid Bing Maps Key:</Label>
+    <InputText @bind-Value="_bingMapsKey"></InputText>
+    <button @onclick="AddRemoveLayer">
+        @(_bingMapsLayer is null ? "Add Bing Maps Basemap Layer" : "Remove Bing Maps Layer")
+    </button>
+</div>
+
+<div class="form-group">
+    <label>Select an Imagery Style</label>
+    <select @onchange="OnStyleChanged">
+        @foreach (BingImageryStyle style in Enum.GetValues(typeof(BingImageryStyle)).Cast<BingImageryStyle>())
+        {
+            if (style == _style)
+            {
+                <option value="@style" selected>@style</option>
+            }
+            else
+            {
+                <option value="@style">@style</option>
+            }
+        }
+    </select>
+</div>
+
+<MapView @ref="_view" OnViewRendered="OnViewRendered"
+         Longitude="-107.95166" Latitude="43.18083" Zoom="4" Class="map-view">
+    <Map ArcGISDefaultBasemap="arcgis-topographic">
+    </Map>
+</MapView>
+
+@code {
+    private void OnViewRendered()
+    {
+        _viewRendered = true;
+    }
+
+    private async Task AddRemoveLayer()
+    {
+        if (!string.IsNullOrEmpty(_bingMapsKey) && _viewRendered)
+        {
+            if (_bingMapsLayer is null)
+            {
+                _bingMapsLayer = new BingMapsLayer(_bingMapsKey, _style);
+                await _view!.AddLayer(_bingMapsLayer, true);
+            }
+            else
+            {
+                await _view!.RemoveLayer(_bingMapsLayer, true);
+                _bingMapsLayer = null;
+            }
+        }
+    }
+    
+    private async Task OnStyleChanged(ChangeEventArgs a)
+    {
+        _style = Enum.Parse<BingImageryStyle>(a.Value!.ToString()!);
+        if (_bingMapsLayer is not null)
+        {
+            await AddRemoveLayer();
+            await AddRemoveLayer();
```

Comment:
why is this called twice?

Reply:
It's a dual-purpose method, call twice and you get remove, then add a new layer. I'm sure there's a better pattern, but 🤷 it's a demo.